### PR TITLE
Clarify rolling-upgrade docs

### PR DESCRIPTION
**Pull request by DaveCTurner**
Note to upgrade the master-eligible nodes last, and note that
`cluster.initial_master_nodes` should not be set.

### CODE REVIEW

File: docs/reference/upgrade/rolling_upgrade.asciidoc [hunk 1]
```diff
@@ -7,6 +7,13 @@ a time so upgrading does not interrupt service. Running multiple versions of
 not supported, as shards cannot be replicated from upgraded nodes to nodes
 running the older version.
 
+It is best to upgrade the master-ineligible nodes in your cluster first and
+then to upgrade the master-eligible nodes. Once enough of the master-eligible
```
Comment (jrodewig) [inline docs/reference/upgrade/rolling_upgrade.asciidoc]: Nit:
```suggestion
then upgrade master-eligible nodes. Once enough master-eligible
```

**Review by jrodewig — APPROVED** [review]
LGTM.

**Comment by jasontedor** [comment]
Relates #47028

**Review by jasontedor — APPROVED** [review]
LGTM. Thanks for picking this up, we had discussed this quite a lot at an internal team meeting a couple of weeks ago.

**Pull request merged**